### ISSUE TITLE
Simple way to allow all CORS / OPTIONS requests?

### ISSUE THREAD

**Issue by bootrino** [falconry/falcon#1220]
I'm wanting to open the door wide to all CORS requests to my application.

I can see there is an additional Falcon/CORS middleware project but I'm hoping there is some simpler way to just open the door wide with a few lines of code and no need to worry about understanding and implementing the middleware and the various options and fine tuning that it provides.

Any suggestions for how to do this?

**Comment by bootrino** [comment]
Answering my own question:

```
class HandleCORS(object):
    def process_request(self, req, resp):
        if req.method == 'OPTIONS':
            resp.set_header('Access-Control-Allow-Origin', '*')
            resp.set_header('Access-Control-Allow-Methods', '*')
            resp.set_header('Access-Control-Allow-Headers', '*')
            resp.set_header('Access-Control-Max-Age', 1728000)  # 20 days
            raise HTTPStatus(falcon.HTTP_200, body='\n')

wsgi_app = api = falcon.API(middleware=[HandleCORS() ])
```

**Comment by vytas7** [comment]
It seems you have coined a solution that fits your needs :smiling_imp: 

Your question is otherwise actually answered in the FAQ [[1]](https://falcon.readthedocs.io/en/stable/user/faq.html#how-do-i-implement-cors-with-falcon), and the answer is really fairly similar to yours. Note the suggestion handles ``'OPTIONS'`` in a slightly different way (by telling which methods are supported instead of saying "all methods"), but yours would do too.

1. https://falcon.readthedocs.io/en/stable/user/faq.html#how-do-i-implement-cors-with-falcon

**Comment by vytas7** [comment]
Also note that a simple to way to add unsophisticated permissive CORS under consideration to be included with Falcon 2.0: https://github.com/falconry/falcon/issues/1194

**Comment by vytas7** [comment]
Thanks for the tip, @MockArch :+1:

Note though that as of Falcon 3.0+, the [CORS functionality](https://falcon.readthedocs.io/en/stable/api/cors.html) is now included in the framework.
The simplest way of adding a permissive CORS policy is now:
```python
import falcon

# Enable a simple CORS policy for all responses
app = falcon.App(cors_enable=True)
```
(You can also address some more complex use cases by adding an instance of [`CORSMiddleware`](https://falcon.readthedocs.io/en/stable/api/cors.html#falcon.CORSMiddleware) to your app.)
